app/src/app/voting/voting.component.ts

```diff
@@ -67,10 +76,121 @@ export class VotingComponent implements OnInit {
+          })
+          // votes per house
+          let votesByHouse: VoteByHouse = {}
+          let finalData: TableRow[] = [...Object.values(currentQuestionAnswersMap).map(answer => ({dekret: answer}))]
+          combined.votes.forEach(voteSnap => {
+            let answer = voteSnap.payload.val()
+            let answerId = Object.keys(answer)[0]
+            let answerString: string = currentQuestionAnswersMap[answerId]
+            let votedBy: string = votingRightToHouseMap[voteSnap.key]
+            let votes = answer[answerId]
+            let answerIndexInData: number = finalData.findIndex(row => row.dekret === answerString)
+            // either create new entry in row or add votes to current
+            finalData[answerIndexInData].hasOwnProperty(votedBy) ? 
+              finalData[answerIndexInData][votedBy] = finalData[answerIndexInData][votedBy] + votes :
+                finalData[answerIndexInData][votedBy] = votes
+            // add total row
+            let voteInVotesByHouse = Object.keys(votesByHouse).includes(votedBy)
+            voteInVotesByHouse ?
+              votesByHouse[votedBy] = votesByHouse[votedBy] + votes :
+                votesByHouse[votedBy] = votes
+          })
+          // add total per answer
+          finalData = finalData.map((row: TableRow) => {
+            let total = Object.values(row)
+              .map(v => typeof(v) === "number" ? v : undefined)
+              .filter(v => v !== undefined)
+              .reduce((partialSum, a) => partialSum + a, 0)
+            return ({...row, total: total})
+          })
+          // add total for total row
+          finalData.push({
+            ...votesByHouse, 
+            dekret: "Celkem hlasů na rod", 
+            total: Object.values(votesByHouse).reduce((partialSum, a) => partialSum + a, 0)})
+          this.setDisplayedHouses([...new Set(Object.values(votingRightToHouseMap))].sort())
+          this.setDisplayedColumns()
+          this.setData(finalData)
+        }
+      )
+    ).subscribe()
+  }
+
+  setData(data: TableRow[]) {
+    this.data = data
+  }
+
+  setDisplayedHouses(houses: string[]) {
+    this.displayedHouses = houses
+  }
+
+  setDisplayedColumns() {
+    let columns = this.displayedHouses.slice()
+    columns.unshift("dekret")
+    columns.push("total")
+    this.displayedColumns = columns
+  }
+
+}
+
+// TODO taken from swiss/src/app/question-form/question-form.components.ts -> should be refactored to common if to be used
+function findVote(snapshots: AngularFireAction<DatabaseSnapshot<any>>[], votingRightId: string, answerId: string) {
+  let snapshot = snapshots.find((row) => row.key == votingRightId);
+  if (snapshot == null) {
+    return ""
+  }
+  let votes = snapshot.payload.val()[answerId]
+  if (votes == null) {
+    return ""
+  }
+  return votes
 }
 
 interface VotingRight {
   id: string,
   name: string,
   votes: number
 }
+
+interface GenericFirebaseSnapshotMapping {
```

Comment:
What sorcery is this? :) (just curious, probably some TS concept I don't know)

Reply:
:) you probably mean one line below:

`[key: string]: string`

it means that this is key-value pair where key is always string and value also always string. It can be used for generic typing of objects when you don't know always beforehand what is exact "name" of the key (basically for dynamically generate objects). But you want to somehow restrict the types there.